compiler/rustc_metadata/src/rmeta/decoder.rs

```diff
@@ -725,78 +725,125 @@ impl MetadataBlob {
+                            privacy = if is_private { "private" } else { "public" },
+                            linkage = if dylib_dependency_formats.is_empty() {
+                                String::new()
+                            } else {
+                                format!(" linkage {:?}", dylib_dependency_formats[i])
+                            }
+                        )?;
+                    }
+                    write!(out, "\n")?;
                 }
-            )?;
-        }
-        write!(out, "\n")?;
-
-        writeln!(out, "\n=Lang items=")?;
-        for (id, lang_item) in root.lang_items.decode(self) {
-            writeln!(
-                out,
-                "{} = crate{}",
-                lang_item.name(),
-                DefPath::make(LOCAL_CRATE, id, |parent| root
-                    .tables
-                    .def_keys
-                    .get(self, parent)
-                    .unwrap()
-                    .decode(self))
-                .to_string_no_crate_verbose()
-            )?;
-        }
-        for lang_item in root.lang_items_missing.decode(self) {
-            writeln!(out, "{} = <missing>", lang_item.name())?;
+
+                "lang_items" => {
+                    writeln!(out, "\n=Lang items=")?;
+                    for (id, lang_item) in root.lang_items.decode(self) {
+                        writeln!(
+                            out,
+                            "{} = crate{}",
+                            lang_item.name(),
+                            DefPath::make(LOCAL_CRATE, id, |parent| root
+                                .tables
+                                .def_keys
+                                .get(self, parent)
+                                .unwrap()
+                                .decode(self))
+                            .to_string_no_crate_verbose()
+                        )?;
+                    }
+                    for lang_item in root.lang_items_missing.decode(self) {
+                        writeln!(out, "{} = <missing>", lang_item.name())?;
+                    }
+                    write!(out, "\n")?;
+                }
+
+                "features" => {
+                    writeln!(out, "\n=Lib features=")?;
+                    for (feature, since) in root.lib_features.decode(self) {
+                        writeln!(
+                            out,
+                            "{}{}",
+                            feature,
+                            if let Some(since) = since {
+                                format!(" since {since}")
+                            } else {
+                                String::new()
+                            }
+                        )?;
+                    }
+                    write!(out, "\n")?;
+                }
+
+                _ => {
+                    writeln!(out, "unknown -Zls kind. allowed values are: no, all, root, lang_items, features")?;
```

Comment:
It doesn't look to me like `no` is handled anywhere

Reply:
Right, that is a leftover from a previous version.